### PR TITLE
Finalize set-up of GitHub actions for building wheels

### PR DESCRIPTION
**Pull request by astrofrog**
### Description

Companion to https://github.com/astropy/astropy/pull/13298 (the present PR should be merged first I believe)

Also fixes https://github.com/astropy/astropy/issues/13194 by not uploading .dev tags (thanks @ConorMacBride!)

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by astrofrog** [comment]
We might want to wait 24h to check that the latest dev wheels upload correctly

**Comment by pllim** [comment]
Please ping me again when you have checked the wheels. Thanks!

**Comment by astrofrog** [comment]
@pllim - they have uploaded so this is good to go, thanks!

**Comment by astrofrog** [comment]
Once this is merged I will email astropy-dev with the new URL

File: docs/install.rst [hunk 1]
```diff
@@ -413,16 +413,16 @@ Installing pre-built Development Versions of ``astropy``
...
 This is useful if you want to test against a development version of astropy but
 do not want to have to build it yourselves. You can see the
-`available astropy dev snapshots page <https://dev.azure.com/astropy-project/astropy/_packaging?_a=package&feed=nightly&package=astropy&protocolType=PyPI&view=versions>`_
+`available astropy dev snapshots page <https://anaconda.org/astropy/astropy/files?type=pypi>`_
 to find out what is currently being offered.
 
 Installing these "nightlies" of ``astropy`` can be achieved by using ``pip``::
 
-  $ pip install --extra-index-url=https://pkgs.dev.azure.com/astropy-project/astropy/_packaging/nightly/pypi/simple/ --pre astropy
+  $ pip install -i https://pypi.anaconda.org/astropy/simple astropy --pre
```
Comment (pllim) [inline docs/install.rst:421]: Do we want to also recommend a few more flags like this?

https://github.com/astropy/astropy/blob/3890fd1b770369e67e33e356f35678a7451cc6fc/tox.ini#L113

Comment (astrofrog) [reply docs/install.rst:421]: I don't think we need those in the instructions? In particular ``--only-binary :all: `` won't be useful as there are only wheels anyway?

Comment (ConorMacBride) [reply docs/install.rst]: `--pre` will also install pre-release versions of any dependencies if they happen to be the latest available on PyPI. I think `--pre` should be fine here as the astropy channel only has astropy, but in sunpy we also use the scipy-wheels-nightly channel so we specify dev dependencies as `astropy>=0.0.dev0` or `matplotlib>=0.0.dev0` etc so we only get the dev version for specific packages and not their dependencies. https://github.com/sunpy/sunpy/blob/main/tox.ini

**Review by pllim — APPROVED** [review]
:shipit: 

Thanks!

**Comment by astrofrog** [comment]
@MeeseeksDev backport to v5.0.x

**Comment by astrofrog** [comment]
@MeeseeksDev backport to v5.1.x

**Pull request merged**